cmd/flipt-edge/main.go

```diff
@@ -0,0 +1,431 @@
+	if err == nil {
+		return userConfigFile, true
+	}
+
+	if !errors.Is(err, fs.ErrNotExist) {
+		defaultLogger.Warn("unexpected error checking configuration file", zap.String("config_file", userConfigFile), zap.Error(err))
+	}
+
+	// finally, check if default config file exists on filesystem
+	_, err = os.Stat(defaultConfigFile)
+	if err == nil {
+		return defaultConfigFile, true
+	}
+
+	if !errors.Is(err, fs.ErrNotExist) {
+		defaultLogger.Warn("unexpected error checking configuration file", zap.String("config_file", defaultConfigFile), zap.Error(err))
+	}
+
+	return "", false
+}
+
+func buildConfig(ctx context.Context) (*zap.Logger, *config.Config, error) {
+	path, found := determineConfig(providedConfigFile)
+
+	// read in config if it exists
+	// otherwise, use defaults
+	res, err := config.Load(ctx, path)
+	if err != nil {
+		return nil, nil, fmt.Errorf("loading configuration: %w", err)
+	}
+
+	if !found {
+		defaultLogger.Info("no configuration file found, using defaults")
+	}
+
+	cfg := res.Config
+
+	encoding := defaultEncoding
+	encoding.TimeKey = cfg.Log.Keys.Time
+	encoding.LevelKey = cfg.Log.Keys.Level
+	encoding.MessageKey = cfg.Log.Keys.Message
+
+	loggerConfig := loggerConfig(encoding)
+
+	// log to file if enabled
+	if cfg.Log.File != "" {
+		loggerConfig.OutputPaths = []string{cfg.Log.File}
+	}
+
+	// parse/set log level
+	loggerConfig.Level, err = zap.ParseAtomicLevel(cfg.Log.Level)
+	if err != nil {
+		defaultLogger.Warn("parsing log level, defaulting to INFO", zap.String("level", cfg.Log.Level), zap.Error(err))
+		// default to info level
+		loggerConfig.Level = zap.NewAtomicLevelAt(zap.InfoLevel)
+	}
+
+	if cfg.Log.Encoding > config.LogEncodingConsole {
+		loggerConfig.Encoding = cfg.Log.Encoding.String()
+
+		// don't encode with colors if not using console log output
+		loggerConfig.EncoderConfig.EncodeLevel = zapcore.CapitalLevelEncoder
+	}
+
+	logger := zap.Must(loggerConfig.Build())
+
+	// print out any warnings from config parsing
+	for _, warning := range res.Warnings {
+		logger.Warn("configuration warning", zap.String("message", warning))
+	}
+
+	if found {
+		logger.Debug("configuration source", zap.String("path", path))
+	}
+
+	return logger, cfg, nil
+}
+
+const (
+	dntVar = "DO_NOT_TRACK"
+	ciVar  = "CI"
+)
+
+func isSet(env string) bool {
+	return os.Getenv(env) == "true" || os.Getenv(env) == "1"
+}
+
+func run(ctx context.Context, logger *zap.Logger, cfg *config.Config) error {
+	isConsole := cfg.Log.Encoding == config.LogEncodingConsole
+
+	if isConsole {
+		color.Cyan("%s\n", banner)
+	} else {
+		logger.Info("flipt starting", zap.String("version", version), zap.String("commit", commit), zap.String("date", date), zap.String("go_version", goVersion))
+	}
+
+	var (
+		isRelease   = release.Is(version)
+		releaseInfo release.Info
+		err         error
+	)
+
+	if cfg.Meta.CheckForUpdates && isRelease {
+		logger.Debug("checking for updates")
+
+		releaseInfo, err = release.Check(ctx, version)
```

Comment:
we'll have to adjust this logic so that it checks for a new release of edge and not Flipt itself

Reply:
Reflecting on it a bit, I am going to remove the check altogether for now.
Doing a release check for something other than the core repo release is a little more fiddly.
We don't have the "latest" flag and can't use the `GetLatestRelease` API call.

I think (at-least near term) this (flipt-edge) is mostly going to be useful when distributed via a docker container only. We're not likely to publish e.g. mac builds of this, or create brew tap and so on, so the GH release might not be that useful. I don't think we need a little update reminder in the Git logs either, and there is no UI to present this information in either.